src/overlays/actors/ovl_Door_Spiral/z_door_spiral.c

```diff
@@ -31,30 +40,221 @@ const ActorInit Door_Spiral_InitVars = {
+    SpiralSceneInfo* spiralSceneInfo = sSpiralSceneInfoTable;
+    s32 i;
+    u8 var_v1;
+
+    for (i = 0; i < ARRAY_COUNT(sSpiralSceneInfoTable); i++) {
+        if (play->sceneId == spiralSceneInfo->sceneId) {
+            break;
+        }
+        spiralSceneInfo += 0x1;
+    }
+    if (i < ARRAY_COUNT(sSpiralSceneInfoTable)) {
+        var_v1 = spiralSceneInfo->index;
+    } else {
+        var_v1 = (Object_GetIndex(&play->objectCtx, GAMEPLAY_DANGEON_KEEP) >= 0) ? 1 : 0;
+    }
+    return var_v1;
+}
+
+void DoorSpiral_Init(Actor* thisx, PlayState* play) {
+    DoorSpiral* this = THIS;
+    s32 transitionActorId = DOORSPIRAL_GET_FC00(thisx);
+    s8 temp_v0_2;
+
+    if (this->actor.room != play->doorCtx.transitionActorList[transitionActorId].sides[0].room) {
+        Actor_Kill(&this->actor);
+    } else {
+        Actor_ProcessInitChain(&this->actor, sInitChain);
+        this->unk145 = DOORSPIRAL_GET_TYPE(thisx);
+        this->unk146 = DOORSPIRAL_GET_80(thisx);
+        this->unk147 = func_809A2BF8(play);
+        temp_v0_2 = Object_GetIndex(&play->objectCtx, sSpiralObjectInfoTable[this->unk147].objectId);
+        this->objIndex = temp_v0_2;
+        if (temp_v0_2 < 0) {
+            Actor_Kill(&this->actor);
+        } else {
+            func_809A2B60(this, func_809A2DB0);
+            Actor_SetFocus(&this->actor, 60.0f);
+        }
+    }
+}
+
+void DoorSpiral_Destroy(Actor* thisx, PlayState* play) {
+    s32 transitionActorId = DOORSPIRAL_GET_FC00(thisx);
+    play->doorCtx.transitionActorList[transitionActorId].id = -play->doorCtx.transitionActorList[transitionActorId].id;
+}
+
+void func_809A2DB0(DoorSpiral* this, PlayState* play) {
+    if (Object_IsLoaded(&play->objectCtx, this->objIndex) != 0) {
+        this->actor.objBankIndex = this->objIndex;
+        func_809A2B70(this, play);
+    }
+}
+
+f32 func_809A2E08(PlayState* play, DoorSpiral* this, f32 arg2, f32 arg3, f32 arg4) {
+    Player* player = GET_PLAYER(play);
+    Vec3f point;
+    Vec3f offset;
+
+    point.x = player->actor.world.pos.x;
+    point.y = player->actor.world.pos.y + arg2;
+    point.z = player->actor.world.pos.z;
+
+    Actor_OffsetOfPointInActorCoords(&this->actor, &offset, &point);
+
+    if ((arg3 < fabsf(offset.x)) || (arg4 < fabsf(offset.y))) {
+        return FLT_MAX;
+    } else {
+        return offset.z;
+    }
+}
+
+s32 func_809A2EA0(DoorSpiral* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    if (Player_InCsMode(play) == 0) {
+        SpiralInfo* temp_v0 = &sSpiralInfoTable[this->unk148];
+        f32 temp_fv0 = func_809A2E08(play, this, 0.0f, temp_v0->unk_0B, temp_v0->unk_0C);
+        if (fabsf(temp_fv0) < 64.0f) {
+            s16 var_v0 = player->actor.shape.rot.y - this->actor.shape.rot.y;
+            if (temp_fv0 > 0.0f) {
+                var_v0 = 0x8000 - var_v0;
+            }
+            if (ABS_ALT(var_v0) < 0x3000) {
+                return (temp_fv0 >= 0.0f) ? 1.0f : -1.0f;
```

Comment:
This function returns `s32`. Are you sure `1.0f` and `-1.0f` are floats?

Reply:
in door_shutter is has a identical function (func_808A0E28) that has the same Ternary

Reply:
Looking at the m2c output it does use floats for this
```c
            if (var_v1 < 0x3000) {
                if (temp_fv0 >= 0.0f) {
                    return (s32) 1.0f;
                }
                return (s32) -1.0f;
            }
            goto block_14;
```